cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/Provider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto.sourcecontrol;
+
+/**
+ * The Provider Enum.
+ */
+public enum Provider {
+  GITHUB;
```

Comment:
The provider should be a string, as it will be whatever the SPI implementation has called itself.

Reply:
if it's not pluggable and we only have 1 provider, we can remove this entirely

Reply:
We support Github in MVP. Will eventually support Gitlab, Bitbucket and Cloud Source Repository. I think this field is more informational, not being used practically yet?